options/options.cc

```diff
@@ -482,6 +482,11 @@ Options* Options::DisableExtraChecks() {
   // By current API contract, not including
   // * verify_checksums
   // because checking storage data integrity is a more standard practice.
+
+  BlockBasedTableOptions* table_options =
+      table_factory->GetOptions<BlockBasedTableOptions>();
+  table_options->detect_filter_construct_corruption = false;
```

Comment:
Hey @pdillinger, I am learning more about how `Options::DisableExtraChecks()` is used. Is opening/reopening the db with the modified option from calling Options::DisableExtraChecks() a must in order for Options::DisableExtraChecks() to take effect? This somehow relates to my question below for Mark.

Reply:
Hi @mrambacher :) I want to modify one particular field `BlockBasedTableOptions::detect_filter_construct_corruption` in the current option `this` but can't think of ways other than modifying through the raw pointer, which does not seem to be a standard practice to me based on the [API doc ](https://github.com/facebook/rocksdb/blob/6.29.fb/include/rocksdb/configurable.h#L66-L69) and codebase search. It seems to me that if we always open/reopen the db with the option containing the modified BlockBasedTableOptions after modification (that is, YES to my question above to @pdillinger ), then it's fine to modify the raw pointer based on the API doc(is it?). 

If not, is there any other way of achieving my goal?

Reply:
Found myself a potential alternative is to make detect_filter_construct_corruption mutable like https://github.com/facebook/rocksdb/pull/8620 and dynamically set it through SetOptions()